extensions/replication/utils/BucketQueueEntry.js

```diff
@@ -0,0 +1,45 @@
+const { splitter } = require('arsenal').constants;
+
+class BucketQueueEntry {
+
+    /**
+     * @constructor
+     * @param {string} key - entry key in database
+     */
+    constructor(key) {
+        const [ownerCanonicalID, bucketName] = key.split(splitter);
+        this._ownerCanonicalID = ownerCanonicalID;
+        this._bucket = bucketName;
+    }
+
+    clone() {
+        return new BucketQueueEntry(this._bucket);
```

Comment:
Instead of `this._bucket`, don't we need to pass a replica of the key here?  Even passing just bucket name will set the clones `ownerCanonicalID` as the bucket name.

Reply:
You're right, good catch. Since this `clone()` function is never used I will just remove it instead.